ons-spark/ancillary-topics/pandas-udfs.ipynb

```diff
@@ -314,92 +333,99 @@
-       "    </tr>\n",
-       "    <tr>\n",
-       "      <th>8</th>\n",
-       "      <td>4306091</td>\n",
-       "      <td>4.25</td>\n",
-       "    </tr>\n",
-       "    <tr>\n",
-       "      <th>9</th>\n",
-       "      <td>4715091</td>\n",
-       "      <td>4.25</td>\n",
+       "      <td>Rabbit</td>\n",
+       "      <td>554650</td>\n",
+       "      <td>192350</td>\n",
+       "      <td>532066</td>\n",
+       "      <td>180010</td>\n",
+       "      <td>25.735436</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "text/plain": [
-       "  incident_number  minute_cost(incident_duration, incident_cost)\n",
-       "0          139091                                           4.25\n",
-       "1          275091                                           4.25\n",
-       "2         2075091                                           4.25\n",
-       "3         2872091                                           4.25\n",
-       "4         3553091                                           4.25\n",
-       "5         3742091                                           4.25\n",
-       "6         4011091                                           4.25\n",
-       "7         4211091                                           4.25\n",
-       "8         4306091                                           4.25\n",
-       "9         4715091                                           4.25"
+       "  incident_number animal_group  easting_rounded  northing_rounded  \\\n",
+       "0          139091          Dog           532350            170050   \n",
+       "1          275091          Fox           534750            167550   \n",
+       "2         2075091          Dog           528050            164950   \n",
+       "3         2872091        Horse           504650            190650   \n",
+       "4         3553091       Rabbit           554650            192350   \n",
+       "\n",
+       "   lfb_easting  lfb_northing   distance  \n",
+       "0       532066        180010   9.964048  \n",
+       "1       532066        180010  12.745802  \n",
+       "2       532066        180010  15.586271  \n",
+       "3       532066        180010  29.408275  \n",
+       "4       532066        180010  25.735436  "
       ]
      },
-     "execution_count": 28,
+     "execution_count": 9,
      "metadata": {},
      "output_type": "execute_result"
     }
    ],
    "source": [
     "@pandas_udf(T.DoubleType(), F.PandasUDFType.SCALAR)\n",
-    "def minute_cost(duration:pd.Series, incidentcost:pd.Series) -> pd.Series:\n",
-    "    duration_minutes = duration*60\n",
-    "    return incidentcost/duration_minutes\n",
-    "\n",
-    "incident_minute_cost = rescue.select('incident_number', minute_cost('incident_duration', 'incident_cost'))\n",
+    "def distx(easting_rounded:pd.Series, lfb_easting:pd.Series, northing_rounded:pd.Series, lfb_northing:pd.Series)->pd.Series:\n",
```

Comment:
I like the example - however I'm still wondering why we need to use a UDF for this? I just played around with this and was able to achieve the same result with this PySpark code: 

lfb_distance_pyspark = rescue.withColumn("pyspark_distance", (((F.col("easting_rounded") - F.col("lfb_easting")) ** 2 + (F.col("northing_rounded") - F.col("lfb_northing")) ** 2) ** 0.5) / 1000)

I know you spent a long time thinking of an appropriate example we can use a UDF for, so I wouldn't say go back to the drawing board. But I wonder if it's worth adding a little disclaimer just saying "Although these examples can be done in PySpark, this example was used just to give an illustration of how to use a pandas UDF. Using a pandas UDF isn't necessarily the most efficient way to do this" or something along those lines.